### PR TITLE
Added some clarity to kubeconfig

### PR DESCRIPTION
**Pull request by jayunit100**
Some things I learned today.  If this is useful to other folks, maybe we should merge it.  Otherwise happy to close this PR.

### CODE REVIEW

**Comment by jayunit100** [comment]
FYI @timothysc first step in sort of clarifying how this works, of course we could use diagrams also for the code path for auth and so on later.

**Comment by mikedanese** [comment]
@kubernetes/goog-ux

File: docs/user-guide/kubeconfig-file.md [hunk 1]
```diff
@@ -83,6 +94,23 @@ users:
     client-key: path/to/my/client/key
 ```
 
+### Building your own kubeconfig file
+
+You can easily use this file as a template to create your own kubeconfig files.  First, lets explain some of the basics.
```
Comment (jlowdermilk) [inline docs/user-guide/kubeconfig-file.md]: We should put a notice in here that if you are deploying k8s via `kube-up.sh`, you don't need to create your own kubeconfig files, the script will do it for you.

**Comment by jlowdermilk** [comment]
@RichieEscarez, ptal

File: docs/user-guide/kubeconfig-file.md (hunk 1 above)
Comment (jayunit100) [inline docs/user-guide/kubeconfig-file.md]: updated and rebase/pushed.  good point.

Comment (pmorie) [inline docs/user-guide/kubeconfig-file.md]: @deads2k

**Comment by jlowdermilk** [comment]
Travis failure is for autogenerated doc munging that needs update. Run `hack/update-generated-docs.sh` to fix.

**Comment by RichieEscarez** [comment]
FYI: ill take a look at this first thing tomorrow

**Comment by jayunit100** [comment]
@RichieEscarez okay if we merge this guy first?  Then 'ill help you do a later PR with your other edits?  I actually have several loose threads at the moment and would like to get this incremental update in....  I think the rewrite you made in the above PR will be super useful as well.

**Comment by timothysc** [comment]
+1
/cc @liggitt for quick turn-around.

**Comment by RichieEscarez** [comment]
@jayunit100 yeah sounds good. I just closed the other PR. I can open new PR against the docs/user-guide/kubeconfig-file.md file once this PR is merged.

**Comment by jayunit100** [comment]
@RichieEscarez ping me when you open ill help you review it ! thanks again for helping improve this!

**Comment by liggitt** [comment]
Same here, there were some tweaks I'd like

**Pull request merged**